src/hotspot/share/nmt/mallocTracker.hpp

```diff
@@ -176,16 +178,12 @@ class MallocMemorySnapshot : public ResourceObj {
   // Total malloc'd memory used by arenas
   size_t total_arena() const;
 
-  void copy_to(MallocMemorySnapshot* s) {
-     // Need to make sure that mtChunks don't get deallocated while the
-     // copy is going on, because their size is adjusted using this
-     // buffer in make_adjustment().
-     ThreadCritical tc;
-     s->_all_mallocs = _all_mallocs;
-     for (int index = 0; index < mt_number_of_types; index ++) {
-       s->_malloc[index] = _malloc[index];
-     }
-   }
+  inline size_t thread_count() const {
+    MallocMemorySnapshot* s = const_cast<MallocMemorySnapshot*>(this);
+    return s->by_type(mtThreadStack)->malloc_count();
+  }
```

Comment:
Why is this needed?

Reply:
It comes after merge, added in JDK-8315362.

Reply:
Yes, JDK-8315362 threw this out. IIRC we removed the awkward mechanism of counting threads by number of thread stacks. Hence my question, why the re-introduction? Merge glitch?

Reply:
It's a merge glitch.